docs/source/configuration/validation.md

```diff
@@ -0,0 +1,204 @@
+The first element should be the name of the `widget`, and the second you can set it up to identify the validator.
+You should specify the `widget` in the JSON schema of the block (or as additional data in the content type definition).
+The next example is for the use case of a block JSON schema:
+
+```ts
+let blockSchema = {
+  // ... fieldset definition in here
+  properties: {
+    ...schema.properties,
+    phone: {
+      title: 'Phone number',
+      description: '',
+      widget: 'phoneNumber',
+    },
+  },
+  required: [],
+};
+```
+
+### Per behavior and field name validator
+
+These validators are applied depending on the behavior (usually coming from a content type definition) in combination with the name of the field.
+
+```ts
+config.registerComponent({
+  name: 'fieldValidator',
+  dependencies: ['plone.eventbasic', 'start', 'startValidator'],
+  component: startEventDateRangeValidator,
+});
+```
+
+The first dependency should be the name of the behavior, and the second the name (`id`) of the field.
+It can get a third dependency in case you want to specify several validators for the same behavior - field id combination.
+This type of validator only applies to content-type validators.
+
+### Per block type and field name validator
+
+These validators are applied depending on the block type in combination with the name of the field in the block settings JSON schema.
+
+```ts
+config.registerComponent({
+  name: 'fieldValidator',
+  dependencies: ['slider', 'url', 'isURL'],
+  component: urlValidator,
+});
+```
+
+The first dependency should be the `id` of the block, and the second the `id` of the field.
+It can get a third dependency in case you want to specify several validators for the same block type - field id combination.
+This type of validator only applies to blocks.
+
+### Specific validator using the `validator` key in the field
+
+A final type of validator is applied to the field if the `validator` key is present in the JSON schema definition of the form field.
+
+```ts
+config.registerComponent({
+  name: 'fieldValidator',
+  dependencies: ['isURL'],
+  component: urlValidator,
+});
+```
+
+The dependencies take one single name, in this case, the name of the validator.
+You should specify the validator in the JSON schema of the block (or as additional data in the content type definition).
+
+```ts
+let blockSchema = {
+  // ... fieldset definition in here
+  properties: {
+    ...schema.properties,
+    customField: {
+      title: 'Default field',
+      description: '',
+      validator: 'isURL',
+    },
+  },
+  required: [],
+};
+```
+
+It does not need to be tied to any field `type` or `widget` definition.
+It runs in addition to all the above, so it complements the other validators if any apply.
+
+## Volto's default validators
+
+Volto provides a set of validators by default, you can find them in this module: `packages/volto/src/config/validators.ts`
```

Comment:
````suggestion
Volto provides a set of validators.
You can find them in the module {file}`packages/volto/src/config/validators.ts`.

```{todo}
Document the API as a reference in another file, after the dust settles.
```
````